tracer/src/Datadog.Tracer.Native/debugger_method_rewriter.cpp

```diff
@@ -837,7 +1250,25 @@ HRESULT DebuggerMethodRewriter::Rewrite(RejitHandlerModule* moduleHandler,
     mdToken exceptionToken = mdTokenNil;
     mdToken callTargetReturnToken = mdTokenNil;
     ILInstr* firstInstruction;
-    hr = debuggerTokens->ModifyLocalSigAndInitialize(&rewriterWrapper, caller, &callTargetStateIndex, &exceptionIndex,
+
+    auto isAsyncMethodProbeHr = IsAsyncMethodProbe(module_metadata.metadata_import, caller);
```

Comment:
I find it a bit confusing that we're determining if a method is async for the second time.
When we're instrumenting a method it's after we did the heavy lifting of determining if a method the user requested to instrument is a kickoff of an async method, and found the `MoveNext` to request rejit for. Here, you're redetermining if you have an async method in hand.
Maybe you can keep aside a bool that will tell us if a method is async (perhaps under `DebuggerRejitHandlerModuleMethod`) and use that here?

Reply:
Internal ticket ID 1051

Reply:
Rethinking about it a bit more, there is an advantage to go with your approach in case the user will choose to instrument `MoveNext` (for some reason). The fact the instrumentation process is not relying on anything to tell something about the method is a plus.
Let's discuss it in-person and decide what path we would like to take in the future.